addons/mail/static/src/web/messaging_menu/notification_item.xml

```diff
@@ -10,8 +10,8 @@
             </div>
             <div class="d-flex flex-column flex-grow-1 align-self-start m-2 overflow-auto">
                 <div class="d-flex">
-                    <span class="o-mail-NotificationItem-name fw-bold text-truncate" t-esc="props.displayName"/>
-                    <span class="o-mail-NotificationItem-counter mx-1 fw-bold" t-if="props.count > 1"> (<t t-esc="props.count"/>)</span>
+                    <span class="o-mail-NotificationItem-name text-truncate" t-att-class="props.hasMarkAsReadButton ? 'fw-bolder' : 'fw-bold'" t-esc="props.displayName"/>
+                    <span t-if="props.counter > 0" class="o-mail-NotificationItem-counter mx-1" t-att-class="props.hasMarkAsReadButton ? 'fw-bolder' : fw-bold"> (<t t-esc="props.counter"/>)</span>
```

Comment:
You removed the logic with undefined after all?
Shouldn't it stay in place for the failure? Or undefined could be removed from there in parent template too.

Reply:
We don't want to show counter of 0 in all cases, and for needaction specifically we don't want to show with `1`.
I can enforce counter of 0 just to avoid `undefined > 0`